### PR TITLE
refactor(apig/channel): use the new channel resource instead

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Using the new channel resource to ireplace the old one.
The new channel resource support theae new functions:
- type control.
- member group.
- more member parameters.
- health check block.
- microservice configuration (only CCE is support yet).

And there are some changes in the new resource parameters:
- rename `algorithm` to `balance_strategy`.
- update the schema type of the parameter `member_type` to **Int**.
- deprecate some old health check parameters.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. deprecate the old VPC channel resource.
2. support the new channel resource.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
PENDING
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccChannel_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccChannel_ -timeout 360m -parallel 4
=== RUN   TestAccChannel_basic
=== PAUSE TestAccChannel_basic
=== RUN   TestAccChannel_eipMembers
=== PAUSE TestAccChannel_eipMembers
=== CONT  TestAccChannel_basic
=== CONT  TestAccChannel_eipMembers
--- PASS: TestAccChannel_eipMembers (509.85s)
--- PASS: TestAccChannel_basic (618.15s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig    618.238s
```

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/deprecated' TESTARGS='-run=TestAccVpcChannel_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/deprecated -v -run=TestAccVpcChannel_basic -timeout 360m -parallel 4
=== RUN   TestAccVpcChannel_basic
=== PAUSE TestAccVpcChannel_basic
=== CONT  TestAccVpcChannel_basic
--- PASS: TestAccVpcChannel_basic (620.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/deprecated      621.036s
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_channel.go [hunk 1]
```diff
@@ -0,0 +1,701 @@
...
+										Description: "schema:Deprecated; The microservice engine ID.",
+									},
+									"service_id": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "schema:Deprecated; The microservice ID.",
+									},
+								},
+							},
+							Description: "schema:Deprecated; The CSE microservice details.",
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apig/resource_huaweicloud_apig_channel.go]:
```suggestion
							Description: "schema:Internal; The CSE microservice details.",
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_channel.go [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.ApigV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	channelId := d.Id()
+	resp, err := channels.Get(client, instanceId, d.Id())
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apig/resource_huaweicloud_apig_channel.go]:
```suggestion
	resp, err := channels.Get(client, instanceId, channelId)
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_channel.go [hunk 3]
```diff
@@ -0,0 +1,701 @@
...
+	if err = channels.Delete(client, instanceId, channelId); err != nil {
+		return diag.Errorf("error deleting APIG channel (%s): %s", channelId, err)
+	}
+
+	return nil
+}
+
+func resourceChannelResourceImportState(_ context.Context, d *schema.ResourceData,
+	_ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), "/", 2)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apig/resource_huaweicloud_apig_channel.go]:
```suggestion
	parts := strings.Split(d.Id(), "/", 2)
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_channel.go [hunk 4]
```diff
@@ -0,0 +1,701 @@
...
+		},
+	}
+	return result
+}
+
+func flattenChannelMicroserivceConfig(microserviceConfig *channels.MicroserviceConfig) []map[string]interface{} {
+	if microserviceConfig == nil {
+		return nil
+	}
+	result := make([]map[string]interface{}, 1)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apig/resource_huaweicloud_apig_channel.go]:
```suggestion
	result := make([]map[string]interface{}, 0, 1)
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_channel.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_channel.go]: fixed.

File: huaweicloud/services/apig/resource_huaweicloud_apig_channel.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_channel.go]: fixed.

File: huaweicloud/services/apig/resource_huaweicloud_apig_channel.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_channel.go]: done.

File: huaweicloud/services/apig/resource_huaweicloud_apig_channel.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_channel.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccChannel_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccChannel_ -timeout 360m -parallel 4
=== RUN   TestAccChannel_basic
=== PAUSE TestAccChannel_basic
=== RUN   TestAccChannel_eipMembers
=== PAUSE TestAccChannel_eipMembers
=== CONT  TestAccChannel_basic
=== CONT  TestAccChannel_eipMembers
--- PASS: TestAccChannel_eipMembers (499.03s)
--- PASS: TestAccChannel_basic (558.91s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      558.981s
```

**Pull request merged**